src/main/java/com/redhat/darcy/webdriver/WebDriverBrowser.java

```diff
@@ -176,8 +178,15 @@ public void closeAll() {
     }
 
     @Override
-    public File takeScreenshot() {
-        return attemptAndGet(() -> driver.getScreenshotAs(OutputType.FILE));
+    public void takeScreenshot(OutputStream outputStream) {
+        byte[] data = attemptAndGet(() -> driver.getScreenshotAs(OutputType.BYTES));
+        try {
+            outputStream.write(data);
+            outputStream.flush();
+            outputStream.close();
```

Comment:
Okay, last (hopefully) request: this should go in a `finally` block or adapt this to use a try-with-resource, and we should either add a check that close is called to your exception test case or create a new test for it